CodeGeneration/Sources/generate-swift-syntax/GenerateSwiftSyntax.swift

```diff
@@ -189,24 +193,25 @@ struct GenerateSwiftSyntax: ParsableCommand {
     }
   }
 
-  private func generateFile(
-    contents: @autoclosure () -> String,
-    destination: URL,
-    verbose: Bool
-  ) throws {
-    try FileManager.default.createDirectory(
-      atPath: destination.deletingLastPathComponent().path,
-      withIntermediateDirectories: true,
-      attributes: nil
-    )
+}
 
-    if verbose {
-      print("Generating \(destination.path)...")
-    }
-    let start = Date()
-    try contents().write(to: destination, atomically: true, encoding: .utf8)
-    if verbose {
-      print("Generated \(destination.path) in \((Date().timeIntervalSince(start) * 1000).rounded() / 1000)s")
-    }
+fileprivate func generateFile(
+  contents: @autoclosure () -> String,
+  destination: URL,
+  verbose: Bool
+) throws {
```

Comment:
I’m happy with making this a top-level function but just FYI, you could keep this a member of `GenerateSwiftSyntax` if you mark the function as `nonisolated`.

Reply:
It doesn't actually use any members, so IMO a top level fileprivate function makes sense anyway.

Reply:
It turns out you don't have to even mark it as `nonisolated` in the current setup of `run`, it still works. 

I originally made this change because when I tried to mark `run` as async without changing the conformance to the async version of `ParseableCommand`, the compiler error kind of implied this was the problem. That was not the case. 

So I don't have a strong opinion on this, but I'm partial to the top level free function.